src/main/java/com/aws/greengrass/lifecyclemanager/KernelAlternatives.java

```diff
@@ -162,31 +169,69 @@ public boolean isLaunchDirSetup() {
         return Files.isSymbolicLink(getCurrentDir()) && validateLaunchDirSetup(getCurrentDir());
     }
 
+    protected boolean canRecoverMissingLaunchDirSetup()
+            throws IOException, URISyntaxException, PackageLoadingException {
+        /*
+        Try and relink launch dir with the following replacement criteria
+        1. check if current Nucleus execution package is valid
+        2. check if any available Nucleus packages in component store is valid
+        3. fail with DirectoryValidationException if above steps do not satisfy
+         */
+        Path currentNucleusExecutablePath = locateCurrentKernelUnpackDir();
+        if (Files.exists(currentNucleusExecutablePath.resolve(KERNEL_BIN_DIR)
+                .resolve(Platform.getInstance().loaderFilename()))) {
+            logger.atDebug().kv("path", currentNucleusExecutablePath)
+                    .log("Current Nucleus executable is valid, setting up launch dir");
+            relinkInitLaunchDir(currentNucleusExecutablePath, true);
+            return true;
+        }
+
+        List<Path> localNucleusExecutablePaths = componentManager.findValidLocalNucleusPackage();
+        if (!localNucleusExecutablePaths.isEmpty()) {
+            Optional<Path> validNucleusExecutablePath = localNucleusExecutablePaths.stream()
```

Comment:
How many nucleus artifacts exist at any given point including the current one that's looking up these paths? 
Should we be choosing the latest one? 
What if the one chosen bricks the device?

Reply:
There can be only 2 nucleus versions at any point. 
It is not likely that the locally present nucleus versions can brick a device because these versions were running correctly on the device at some point and they were updated due to an OTA update.